test-harness/init.sh

```diff
@@ -77,17 +78,21 @@ function template_build_targets() {
     [[ -z $GIT_DIFF_SOURCEBRANCH ]] && echo "ERROR: GIT_DIFF_SOURCEBRANCH wasn't provided" && return 1
 
     echo "INFO: Running git diff from branch ${GIT_DIFF_SOURCEBRANCH}"
-    files=(`git diff ${GIT_DIFF_UPSTREAMBRANCH} ${GIT_DIFF_SOURCEBRANCH} --name-only|grep -v *.md`)
+    files=(`git diff ${GIT_DIFF_UPSTREAMBRANCH} ${GIT_DIFF_SOURCEBRANCH} --name-only|grep -v *.md||true`)
```

Comment:
should we instead prefer a white-list of file extensions? The list would probably include `.tf` and other terraform related files. May be more robust this way so that the tests won't be re-run if somebody adds, for example, an image or text file.

Reply:
I'm going to add the following 
```
GIT_DIFF_EXTENSION_FILTER="*.md"
files=(`git diff ${GIT_DIFF_UPSTREAMBRANCH} ${GIT_DIFF_SOURCEBRANCH} --name-only|grep -v ${GIT_DIFF_EXTENSION_FILTER}||true`)
```

if you want to support multiple extensions then you can do something like this 
GIT_DIFF_EXTENSION_FILTER="*.md|*.txt"

Reply:
Seems like we care only about when Terraform files change, so why not explicitly filter for those (i.e., `grep` and not `grep -v`? With the current logic, adding a `.txt` file to each directory would cause a massive integration test to run.

I think a white-list approach is less brittle. Though what you have won't break anything, so I suppose its not a blocking comment.

Reply:
That’s a good call out. We’d have to account for changes to docker files or the test harness shell script(s) and go filed... I’m thinking the pattern would look something like this “*.go|*.tf|*.sh|Dockerfile*|*.tfvars”.